File: game.py
from numpy import zeros

class Game(object):
    def __init__(self, width, height):
        self.width = width
        self.height = height
        self.blackPixel = zeros((self.width, self.height), dtype=bool)

    def ajouterPixel(self, antX, antY):
        if(self.blackPixel[antX][antY]==False): 
            self.blackPixel[antX][antY] = True
        else:
            self.blackPixel[antX][antY] = False

    def returnColor(self, x, y):
        return self.blackPixel[x][y]
    
    def exportMap(self):
        return self.blackPixel.tobytes()

    def getSize(self):
        return "{},{}".format(self.width, self.height)

File: game.js
class Game{
    constructor(height, width){
        this.height = height;
        this.width = width;
        this.blackPixel = [];
    }

    ajouterPixel(x, y){
        blackPixel.push(x+y*width);
    }

    returnCase(antPos){
        if(antPos in blackPixel) return false;
        else return true;
    }

    exportMap(){
        /*
            Charge la carte envoyé par le serveur
        
        */
       return null;
    }

    getLastPixel(){
        //retourne le dernier pixel de la liste AKA le dernier rajouter
        if(this.blackPixel.length>0) return null;
        else return this.blackPixel[this.blackPixel.length-1];
    }
}

module.exports = Game;